rust/kernel/sync/arc.rs

```diff
@@ -97,6 +97,35 @@ impl<T> Arc<T> {
         Ok(unsafe { Self::from_inner(inner) })
     }
 
+    /// Tries to allocate a new [`Arc`] instance whose contents are not initialised yet.
+    pub fn try_new_uninit() -> Result<Arc<MaybeUninit<T>>> {
+        let layout = Layout::new::<ArcInner<MaybeUninit<T>>>();
```

Comment:
Please make this code part of `UniqueArc::try_new_uninit`'s implementation. The `Arc::try_new_uninit` API makes no sense, given that unlike std's `Arc`, our `Arc` does not have `get_mut` so there logically an uninitialised `Arc` will be uninitialised forever.